api/src/test/java/org/pmiops/workbench/db/model/StorageEnumsTest.java

```diff
@@ -3,38 +3,51 @@
 import static com.google.common.truth.Truth.assertThat;
 import static com.google.common.truth.Truth.assertWithMessage;
 
+import com.google.api.client.util.Sets;
+import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 import org.junit.Test;
+import org.pmiops.workbench.model.Disability;
 import org.pmiops.workbench.model.Domain;
 
 public class StorageEnumsTest {
   private final Object INDICATES_STATIC_METHOD = null;
 
-  final Set<Class> enumClasses = getEnumerationClasses();
-  final Set<Method> methods = arraysToSet(DbStorageEnums.class.getDeclaredMethods());
-
-  // e.g. public static Short reviewStatusToStorage(ReviewStatus s)
-  final Map<Class, Method> enumClassToStorageMethod =
-      methods.stream()
-          // domainToDomainId is stringly typed - test with test_domainId
-          .filter(m -> !m.getName().equals("domainToDomainId"))
-          .filter(m -> enumClasses.contains(firstMethodParameterType(m)))
-          .collect(Collectors.toMap(this::firstMethodParameterType, m -> m));
-
-  // e.g. public static ReviewStatus reviewStatusFromStorage(Short s)
-  final Map<Type, Method> storageMethodToEnumClass =
-      methods.stream()
-          // domainIdToDomain is stringly typed - test with test_domainId
-          .filter(m -> !m.getName().equals("domainIdToDomain"))
-          .filter(m -> enumClasses.contains(m.getReturnType()))
-          .collect(Collectors.toMap(Method::getReturnType, m -> m));
+  final List<Field> allFields = Arrays.asList(DbStorageEnums.class.getDeclaredFields());
+  final List<Method> allMethods = Arrays.asList(DbStorageEnums.class.getDeclaredMethods());
+
+  @Test
+  public void test_duplicateFieldNames() {
+    Set<String> seenFieldNames = Sets.newHashSet();
+    for (Field field : allFields) {
+      String fieldName = field.getName();
+      assertThat(fieldName).isNotIn(seenFieldNames);
+      seenFieldNames.add(fieldName);
+    }
+  }
+
+  @Test
+  public void test_duplicateMethodNames() {
+    Set<String> seenMethodNames = Sets.newHashSet();
+    for (Method method : allMethods) {
+      String methodName = method.getName();
+      // TODO: sort out the Disability mess: it can be Boolean, Short, or Disability
```

Comment:
working on this.  fixing here would be confusing, I think.

Reply:
wow this looks like a mess. I hope in the follow up PR you're shifting this all to the enum?

Reply:
I expect that this will be possible